src/eureka/S6_planet_spectra/s6_spectra.py

```diff
@@ -423,7 +423,9 @@ def parse_s5_saves(meta, log, fit_methods, channel_key='shared'):
 
         fname = f'S5_{fitter}_samples_{channel_key}'
 
-        keys = [key for key in full_keys if y_param == key[:len(y_param)]]
+        temp_keys = [y_param+f'_{c}' if c > 0 else y_param
+                     for c in range(meta.nspecchan)]
+        keys = [key for key in temp_keys if key in full_keys]
```

Comment:
I think we need to apply a similar fix on Line 451

Reply:
Good point!